lib/rules/no-unused-vars.js

```diff
@@ -623,9 +623,7 @@ module.exports = {
                             if (config.caughtErrorsIgnorePattern && config.caughtErrorsIgnorePattern.test(def.name.name)) {
                                 continue;
                             }
-                        }
-
-                        if (type === "Parameter") {
+                        } else if (type === "Parameter") {
```

Comment:
> all checks all variables for usage, including those in the global scope. This is the default setting

The description in the docs for `vars` is confusing, the intention seems to check all variables. However, the current desired behavior should be that "it only applies to those not covered by other ignore pattern options." Can you update it in the docs too?

Reply:
Done!